layers/+tools/dap/funcs.el

```diff
@@ -0,0 +1,66 @@
+;;; packages.el --- DAP mode functions File for Spacemacs
+;;
+;; Copyright (c) 2012-2018 Sylvain Benner & Contributors
+;;
+;; Author: Ivan Yonchovski (yyoncho@gmail.com)
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defun spacemacs/dap-bind-keys-for-mode (mode)
+  "Define key bindings for the specific MODE."
+
+  (spacemacs/declare-prefix-for-mode mode "md" "debug")
+  (spacemacs/declare-prefix-for-mode mode "mdd" "debuging")
+  (spacemacs/declare-prefix-for-mode mode "mdb" "breakpoints")
+  (spacemacs/declare-prefix-for-mode mode "mdl" "debug windows")
+  (spacemacs/declare-prefix-for-mode mode "mdS" "switch")
+  (spacemacs/declare-prefix-for-mode mode "mdI" "inspect")
+  (spacemacs/declare-prefix-for-mode mode "mde" "eval")
+
+  (spacemacs/set-leader-keys-for-major-mode mode
+    ;; debuging/running
+    "ddd" #'dap-debug
+    "ddl" #'dap-debug-last
+    "ddr" #'dap-debug-recent
+    ;; stepping
+    "dc" #'dap-continue
```

Comment:
I am not yet convinced all these should be without another name-space. It does make typing them easier, but then again we have the hydra...